### ISSUE TITLE
Wrong GST amount charged on single Collective

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#6365]
**Describe the bug**
The GST amounts for these transactions seem to all be wrong: https://opencollective.com/tamaki-makaurau-mas

$10 contribution charging $15 in GST when it should be $1.50, and the same for other amounts. Seems to be like 150% GST instead of 15% maybe?

Further, why are these transactions attracting GST at all? These appear to be normal donations, which should not have GST. Only products and event tickets should have GST applied. 

This only appears to be affecting this one Collective. Is there something broken with how their tiers are set up or something?

Were these erroneous amounts actually charged to the contributors? It's not adding up at all. If so, we'll need to refund this GST money to them. If not, we need to get the platform data corrected.

  P2 low frequency, high impact

**Comment by Betree** [comment]
@alanna All their tiers are set up as "SERVICE", so it's normal that we charged GST. Would be good to understand why they did that.

There's indeed something wrong with the amounts, I'm continuing the investigation.

**Comment by Betree** [comment]
Found 8 orders affected by this issue: 4 with GST, 4 with VAT.

<details>

```sql
select
    ROUND(("totalAmount" - "taxAmount" - COALESCE("platformTipAmount", 0)) * ((data -> 'tax' ->> 'percentage')::numeric / 100)),
    "taxAmount",
    "totalAmount" - "taxAmount" - COALESCE("platformTipAmount", 0) as "netAmount",
    data -> 'tax' ->> 'percentage' as percentage,
     *
from "Orders"
where "taxAmount" IS NOT NULL
AND data -> 'tax' ->> 'percentage' IS NOT NULL
AND "totalAmount" != 0
AND ROUND(("totalAmount" - "taxAmount" - COALESCE("platformTipAmount", 0)) * ((data -> 'tax' ->> 'percentage')::numeric / 100))  != "taxAmount"
ORDER BY "createdAt" DESC
```

</details>

I'm pushing a series of PRs to solve this issue:
- Frontend: The issue is supposedly fixed by the recent changes in the contribution flow
- https://github.com/opencollective/opencollective-api/pull/8453: for the future, to detect & prevent similar issues in the API
- _To be updated_ => to fix existing entries in the ledger

**Comment by alanna** [comment]
@Betree can you confirm if the incorrect rates were actually charged to the user? Do we need to do refunds? 

In the OCNZ case, if we are doing refunds we probably would like to refund the full GST amount as it was incorrectly set up as a service in the first place.

**Comment by Betree** [comment]
@alanna The tax amount displayed in the breakdown was surely wrong, but the **total amount** displayed was supposedly correct: contributors normally know how much they've contributed. Most "total amounts" are round numbers ($100, $30, $10) which tends to confirm that.

**Comment by alanna** [comment]
Unfortunately there is still an issue with these transactions. Despite them claiming to include GST, when I look on Stripe I can see only the ex-GST amount was received (i.e. $10 instead of the expected $11.50).

In this case this is actually good because they added GST to this tier in error, but I'm worried this indicates a larger issue with how the system is not collecting local tax amounts.

The transactions show GST:
<img width="730" alt="Screen Shot 2023-02-13 at 2 04 30 PM" src="https://user-images.githubusercontent.com/5498878/218349397-1311aa16-3eb4-4498-9292-3ae34cc14700.png">

But in Stripe it's not collecting the GST amounts (the $100 one is $80 + $20 platform tip):
<img width="1098" alt="Screen Shot 2023-02-13 at 2 04 20 PM" src="https://user-images.githubusercontent.com/5498878/218349438-492ba582-a934-476a-ace8-7a2c1b3c4131.png">

**Comment by alanna** [comment]
Also, why does it say GST on $10 would be $1.30? It should be $1.50 as it's 15%. Same for the $80 one, it should be $12 not $10.43. This is probably a separate issue?

**Comment by Betree** [comment]
These collective is still "half-migrated", its transactions are recorded with GST but I have a pending task to strip all tax records from them (since they mistakingly created the tiers as "services" while they were not).

Regarding the existence of a broader issue, I'll try and look at it during the week. If you have examples with other collectives and/or with a more recent date that'd be helpful.

**Comment by alanna** [comment]
I don't know if there are any other examples as OCNZ hasn't had any other GST transactions recently, but I suggest ensuring VAT transactions are unaffected.

**Comment by Betree** [comment]
I checked the data quickly and there's nothing new here. The recurring contributions that were broken got their new transactions recorded with these broken values. 

We had a plan for that, but were not able to prioritize it yet. Let's keep the issue open to be clear about what's left to do: 
- [ ] Re-run the script to update tax amount on the transactions
- [ ] Implement a script to remove taxes from existing transactions/orders and apply it to the contributions mentioned above

**Comment by Betree** [comment]
Removed all taxes from contributions to https://opencollective.com/tamaki-makaurau-mas. We now have a script to do it easily if needed: https://github.com/opencollective/opencollective-api/pull/8595.

While investigating one of the two contributions, I found the discrepancy's root cause; we're not updating the tax amount properly when editing the amount of an existing recurring contribution: https://github.com/opencollective/opencollective/issues/6508.
